modules/os2forms_fasit/src/Helper/FasitHelper.php

```diff
@@ -0,0 +1,488 @@
+
+    // Fix if os2forms_person_lookup (cpr & name validation) element is used.
+    if (is_array($fasitCpr)) {
+      // Example:
+      // [
+      // 'cpr_number' => 1234567890,
+      // 'name' => Eksempel Eksempelsen,
+      // ].
+      $fasitCpr = $fasitCpr['cpr_number'] ?? NULL;
+    }
+
+    if (NULL === $fasitCpr) {
+      throw new InvalidSubmissionException(sprintf('Could not determine value of configured CPR element in submission.'));
+    }
+
+    // Construct XML.
+    $doc = new \DOMDocument();
+
+    if (!$doc->load(__DIR__ . '/templateCreateDocument.xml')) {
+      throw new FasitXMLGenerationException('Could not load template XML');
+    }
+
+    // Set Document values.
+    $doc->getElementsByTagName('BeskrivelseTekst')->item(0)->nodeValue = $fasitDocumentDescription;
+    $doc->getElementsByTagName('TitelTekst')->item(0)->nodeValue = $fasitDocumentTitle;
+    $doc->getElementsByTagName('BrevDato')->item(0)->nodeValue = (new \DateTimeImmutable())->format('Y-m-d');
+
+    // Handle uploads
+    // Use existing DelEgenskaber as template for each upload.
+    // DelEgenskaber is the first child of Del.
+    $delEgenskaber = $doc->getElementsByTagName('DelEgenskaber')->item(0);
+    $del = $doc->getElementsByTagName('Del')->item(0);
+
+    // Append a DelEgenskaber per upload.
+    foreach ($uploads as $upload) {
+      $copyDelEgenskaber = $delEgenskaber->cloneNode(TRUE);
+      $copyChildNodes = $copyDelEgenskaber->childNodes;
+      foreach ($copyChildNodes as $childNode) {
+        if ($childNode->nodeName === 'DelTekst') {
+          $childNode->nodeValue = $upload['filename'];
+        }
+        elseif ($childNode->nodeName === 'IndholdTekst') {
+          $childNode->nodeValue = $upload['id'];
+        }
+      }
+
+      $del->insertBefore($copyDelEgenskaber, $delEgenskaber);
+    }
+
+    // Remove template 'DelEgenskaber'.
+    $parent = $delEgenskaber->parentNode;
+    $parent->removeChild($delEgenskaber);
+
+    // Handle Parter.
+    $elements = $doc->getElementsByTagName('Parter')->item(0)->childNodes;
+
+    foreach ($elements as $element) {
+      if ($element->nodeName === 'ReferenceID') {
+        $referenceIdChildElements = $element->childNodes;
+        foreach ($referenceIdChildElements as $referenceIdChildElement) {
+          if ($referenceIdChildElement->nodeName === 'UUIDIdentifikator') {
+            $referenceIdChildElement->nodeValue = self::FASIT_API_PRIMAERPART_TEMPLATE_VALUE . $fasitCpr;
+          }
+        }
+      }
+    }
+
+    [$certificateOptions, $tempCertFilename] = $this->getCertificateOptionsAndTempCertFilename();
+
+    $options = [
+      'headers' => [
+        'Content-Type' => 'application/xml',
+      ],
+      'body' => $doc->saveXML(),
+      'cert' => $certificateOptions,
+    ];
+
+    // Attempt upload.
+    try {
+      $response = $this->client->request('POST', $endpoint, $options);
+    }
+    catch (GuzzleException $e) {
```

Comment:
This again locks the module into using guzzle?

Reply:
See other comment.